src/bundle-manifest.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * A bundle is a grouping of files which serve the need of one or more
+ * entrypoint files.
+ */
+export interface IBundle {
+  entrypoints: Set<URL>;
+  files: Set<URL>;
+}
+
+/**
+ * A bundle strategy function is used to transform an array of bundles.
+ */
+export type IBundleStrategy = (bundles: IBundle[]) => IBundle[];
+
+/**
+ * A mapping of URLs to their full set of transitive dependencies, such that
+ * `a->b, b->c, b->d, e->f` would be represented `{a:[b,c,d], e:[f]}`
```

Comment:
Good comment. Can you add a dependency shared between a and f, and also point out that a and f are the entrypoints, presumably why they're not keys in the map.